bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
 ReminderMention = t.Union[UnambiguousUser, discord.Role]
 
 
+class SnoozeSelectView(discord.ui.View):
+    """The reminder's select dropdown UI View."""
+
+    SNOOZE_DURATIONS: dict[str, t.Optional[int]] = {
+        # Mapping of `duration string: num_seconds_in_duration`.
+        #
+        # `num_seconds_in_duration` is `None` for durations >= 1 month
+        # since the amount of seconds depends on the date of lookup
+        # (i.e. seconds from 1st jan to 1st feb is different than from 1st feb to 1st march etc).
+        "1 minute": 60,
+        "5 minutes": 5 * 60,
+        "15 minutes": 15 * 60,
+        "30 minutes": 30 * 60,
+        "1 hour": 60 * 60,
+        "3 hours": 3 * 60 * 60,
+        "6 hours": 6 * 60 * 60,
+        "1 day": 24 * 60 * 60,
+        "1 week": 7 * 24 * 60 * 60,
+        "1 month": None,
+        "3 months": None,
+        "6 months": None,
+        "1 year": None
+    }
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=30)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.interaction_owner_id = reminder_author_id
+
+        self.dropdown: discord.ui.Select = self.children[0]
+        for duration in self.SNOOZE_DURATIONS:
+            self.dropdown.add_option(label=duration)
```

Comment:
We can add a value here which will be sent by the api. Rather than having to map them all up on receiving them, we can send both what to display and what we want to see to the api.

Reply:
I did notice this, but I also don't see a point in doing so. We only ever use the label, so there's not much point in defining a value (especially since not all values are fixed -- for durations >= 1 month).